lib_eio_luv/eio_luv.ml

```diff
@@ -192,7 +192,8 @@ end = struct
     let m = if Lwt_dllist.is_empty events.read then m else `READABLE :: m in
     if m = [] then (
       Luv.Poll.stop events.handle |> or_raise;
-      t.fd_map <- Fd_map.remove events.fd t.fd_map
+      t.fd_map <- Fd_map.remove events.fd t.fd_map;
+      Luv.Handle.close events.handle (fun () -> ())
```

Comment:
BTW, I just tested with [godbolt](https://ocaml.godbolt.org/) that `Fun.id` instead of `fun () -> ()` or `ignore` generates one instruction less.  `Fun.id` is a function that just returns.  `fun () -> ()` and `ignore` are functions that return the unit value (and that takes an additional instruction).  I could imagine a type directed optimization that would eliminate this difference in case `ignore` is called with the unit value and it should also be possible to optimize `fun () -> ()`, but the compiler doesn't seem to do that yet.

I don't suggest to change the code in this PR, but it is something I might want to consider in performance critical cases.

Reply:
Ah, interesting! I didn't want to use `ignore` here in case `close` changed in future to pass a result type or something, but `Fun.id` would have worked.